internal/dao/dao_errors.go

```diff
@@ -38,4 +38,7 @@ var (
 
 	// ErrReservationRateExceeded is returned when SQL constraint does not allow to insert more reservations
 	ErrReservationRateExceeded = usrerr.New(429, "rate limit exceeded", "too many reservations, wait and retry")
+
+	// ErrPubkeyNotFound is returned when a nil pointer to a pubkey is used for reservation detail
+	ErrPubkeyNotFound = usrerr.New(404, "pubkey not found", "no pubkey found")
```

Comment:
Maybe to stress out the fact that this was caused very likely by deletion (the usecase we see quite often):

```suggestion
	ErrPubkeyNotFound = usrerr.New(404, "pubkey not found", "no pubkey found (already deleted?)")
```

But I have to admit it looks weird, 404 is a 404 period. So feel free to let this wild idea go :-)

Reply:
I have modified the message a bit to put the information out there.

Reply:
One more idea, we might want to incorporate a message telling users what can they do to fix this, right?

Reply:
I thought it cannot be fixed, but sure that is the intention yes.